From: You-Sheng Yang <vicamo@gmail.com>
Date: Fri, 13 Jun 2025 18:46:24 +0800
Subject: wl: use timer_container_of for kernel >= 6.16

`from_timer` macro renamed in v6.16-rc1 commit 41cb08555c416 ("treewide,
timers: Rename from_timer() to timer_container_of()").

Bug-Ubuntu: https://bugs.launchpad.net/bugs/2114261
Signed-off-by: You-Sheng Yang <vicamo@gmail.com>
---
 src/wl/sys/wl_linux.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/wl/sys/wl_linux.c b/src/wl/sys/wl_linux.c
index d36f8bc..8bf80ab 100644
--- a/src/wl/sys/wl_linux.c
+++ b/src/wl/sys/wl_linux.c
@@ -2377,7 +2377,9 @@ wl_timer(
 #endif
 ) {
 	wl_timer_t *t =
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 15, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 16, 0)
+		timer_container_of(t, tl, timer);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 15, 0)
 		from_timer(t, tl, timer);
 #else
 		(wl_timer_t *)data;
